python/tests/test_semantics.py

```diff
@@ -592,6 +592,35 @@ def test_process(f):
     assert p1.status              == 'COMPLETE'
 
 
+def test_path(f):
+    key = fingerprint('PATH', 'PATH1', 10, 0)
+    p1 = f.objects[key]
+
+    key = fingerprint('CHANNEL', 'CHANN1', 10, 0)
+    in1 = f.objects[key]
+
+    key = fingerprint('CHANNEL', 'CHANN2', 10, 0)
+    in2 = f.objects[key]
+
+    key = fingerprint('FRAME', 'FRAME1', 10, 0)
+    fr1 = f.objects[key]
+
+    key = fingerprint('WELL-REFERENCE', 'THE-WELL', 10, 0)
+    wr = f.objects[key]
+
```

Comment:
Should be:

assert p1.tool_zero_offset == -7
assert p1.angular_drift    == Channel('CHANN1', 10, 0)
assert p1.vertical_depth   == Channel('CHANN2', 10, 0)

tool has a whitespace in attributes, angular/vertical needs to be in linkage